src/task.c

```diff
@@ -1112,9 +1112,8 @@ task_handler (gpointer user_data)
       result = G_SOURCE_CONTINUE;
       break;
   }
-  if (message->len) {
-    if (fwrite(message->str, sizeof(gchar), message->len, stderr) != message->len)
-        g_warning ("failed to write message");
+  if (message->len > 0) {
+    g_printerr ("%s", message->str);
```

Comment:
```
g_printerr() should not be used from within libraries. 
Instead g_log() or g_log_structured() should be used,
or the convenience macros g_message(), g_warning() and g_error().
```

I think we spoke about g_printerr that we don't want to use it or?

Reply:
In a different context tho. It was discussed for reporting errors within a function, and the comment made was that, instead of printing it, the error should be returned through GError and the caller should decide what to do with it.

In this case, these are harness messages and they should be printed to restraintd stderr.

I used g_printerr instead of fwrite because we need to supress these messages when restraintd runs with the client. And I didn't use g_message or similar to keep output as it was in 0.1.45, as these macros add,
```
** Message: XX:YY:ZZ.NNN: the message...

```

With a new line too.

Also, regarding fwrite, doesn't make much sense to use it with GStrings, as these guarantee that the str member is going to be NULL-terminated.